### PR TITLE
Mliddell/questionnaire control apl

### PR DESCRIPTION
**Pull request by mikelid**
## Description
- No auto-submit when last question answered by touch (submit reqd)
- Submit button disables further input to prevent chaos
  and a wait indicator shown after a delay in case UPL is very high
- more configuration with i18n defaults

## Motivation and Context
The APL interactions had some issues:
1. if user is allowed to press buttons after submit, the interaction is complicated and confusing to user
2. auto-submission when all questions are filled out is not right (and suffers from #1).
3. no enough APL detail was configurable.

## Testing
questionnaire demo skill, regression tests

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs(Add new document content)
- [ ] Translate Docs(Translate document content)

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://https://github.com/alexa/ask-sdk-controls/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: src/commonControls/questionnaireControl/QuestionnaireControl.ts [hunk 1]
```diff
@@ -1144,10 +1155,30 @@ export class QuestionnaireControl extends Control implements InteractionModelCon
         return;
     }
 
+    private wantsToKeepAplInitiative(input: ControlInput): boolean {
+        const content = this.getQuestionnaireContent(input);
```
Comment (ShenChen-Amazon) [inline src/commonControls/questionnaireControl/QuestionnaireControl.ts:1159]: this content is never used

File: src/commonControls/questionnaireControl/QuestionnaireControl.ts [hunk 2]
```diff
@@ -1144,10 +1155,30 @@ export class QuestionnaireControl extends Control implements InteractionModelCon
...
+        try {
+            okIf(this.inputWasAnswerByTouch);
```
Comment (ShenChen-Amazon) [inline src/commonControls/questionnaireControl/QuestionnaireControl.ts:1161]: more like a question: is it means any touch event will make this control be able to take Initiative ?

File: src/commonControls/questionnaireControl/QuestionnaireControl.ts [hunk 3]
```diff
@@ -1012,7 +1018,7 @@ export class QuestionnaireControl extends Control implements InteractionModelCon
         const questionIndex = this.getQuestionIndexById(content, questionId);
         const choiceId = choiceIndex === -1 ? undefined : content.choices[choiceIndex].id;
         this.updateAnswer(questionId, choiceId, input, resultBuilder);
-        this.addCompletionActIfImplicitlyComplete(input, resultBuilder);
+        //we specifically do _not_ call addCompletionActIfImplicitlyComplete on answer by touch.
```
Comment (ShenChen-Amazon) [inline src/commonControls/questionnaireControl/QuestionnaireControl.ts:1021]: probably move the reason here as well

File: src/commonControls/questionnaireControl/QuestionnaireControl.ts [hunk 4]
```diff
@@ -853,7 +859,7 @@ export class QuestionnaireControl extends Control implements InteractionModelCon
 
             this.addCompletionActIfImplicitlyComplete(input, resultBuilder);
         } else {
-            throw new Error('todo');
+            throw new Error('todo'); //TODO. handle case of a value that doesn't match a legal answer.
```
Comment (ShenChen-Amazon) [inline src/commonControls/questionnaireControl/QuestionnaireControl.ts:862]: probably not good idea to leave the error message as 'TODO'

**Review by ShenChen-Amazon — APPROVED** [review]
Spent some time understanding the questionareControl, focused on the changes by referencing the description. LGTM

File: test/commonControlTests/QuestionnaireControl.spec.ts [hunk 5]
```diff
@@ -167,6 +171,22 @@ suite('QuestionnaireControl e2e tests', () => {
                 choiceId: 'no',
             },
         });
+
+        const response2 = await testTurn(
+            invoker,
+            'U: <Press "yes" for "have have headache">',
```
[the remark this answers is not in the document]
Comment (mikelid) [reply test/commonControlTests/QuestionnaireControl.spec.ts:177]: fixed

File: src/commonControls/questionnaireControl/QuestionnaireControl.ts (hunk 1 above)
Comment (mikelid) [reply src/commonControls/questionnaireControl/QuestionnaireControl.ts:1159]: removed

File: src/commonControls/questionnaireControl/QuestionnaireControl.ts (hunk 2 above)
Comment (mikelid) [reply src/commonControls/questionnaireControl/QuestionnaireControl.ts:1161]: this.inputWasAnswerByTouch will only be true if the input on this turn was specifically a touch event on a radio button of this control.  In this situation we will keep the APL up and not speak a prompt or open the mic.

File: src/commonControls/questionnaireControl/QuestionnaireControl.ts (hunk 3 above)
Comment (mikelid) [reply src/commonControls/questionnaireControl/QuestionnaireControl.ts:1021]: added.

**Pull request merged**